src/vs/workbench/contrib/scm/browser/scmViewPane.ts

```diff
@@ -1182,6 +1182,20 @@ class ViewModel {
 			if (e.reason === WillSaveStateReason.SHUTDOWN) {
 				this.storageService.store(`scm.viewState`, JSON.stringify(this.treeViewState), StorageScope.WORKSPACE, StorageTarget.MACHINE);
 			}
+
+			this.mode = this.getViewModelMode();
+			this.sortKey = this.getViewModelSortKey();
+		});
+
+		this.storageService.onDidChangeValue(e => {
+			switch (e.key) {
```

Comment:
Can we test that `onDidChangeValue` is called when workspace state changes, I am not 100% sure we send out events, because normally this never changes outside the window. 

And if it does, do we possibly apply it twice whenever changing it here?

Reply:
It does get called, and the setter for `this.mode` and `this.sortKey` will check whether the value has changed and bail out if it has not. This ensures that there is not an infinite loop of setting the same value which emits `onDidChangeValue`.